src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedAccessControlService.java

```diff
@@ -0,0 +1,278 @@
+        return super.filterCaseFieldsByAccess(caseFields,
+            caseFieldDefinitions,
+            userRoles,
+            access,
+            isClassification);
+    }
+
+    @Override
+    public List<CaseFieldDefinition> filterCaseFieldsByAccess(List<CaseFieldDefinition> caseFieldDefinitions,
+                                                              Set<String> userRoles,
+                                                              Predicate<AccessControlList> access) {
+        applyAccessProfileRules(caseFieldDefinitions);
+
+        return super.filterCaseFieldsByAccess(caseFieldDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public CaseUpdateViewEvent filterCaseViewFieldsByAccess(CaseUpdateViewEvent caseEventTrigger,
+                                                            List<CaseFieldDefinition> caseFieldDefinitions,
+                                                            Set<String> userRoles,
+                                                            Predicate<AccessControlList> access) {
+        applyAccessProfileRules(caseFieldDefinitions);
+
+        return super.filterCaseViewFieldsByAccess(caseEventTrigger,
+            caseFieldDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public List<AuditEvent> filterCaseAuditEventsByReadAccess(List<AuditEvent> auditEvents,
+                                                              List<CaseEventDefinition> caseEventDefinitions,
+                                                              Set<String> userRoles) {
+        return super.filterCaseAuditEventsByReadAccess(auditEvents,
+            caseEventDefinitions,
+            userRoles);
+    }
+
+    @Override
+    public List<CaseStateDefinition> filterCaseStatesByAccess(List<CaseStateDefinition> caseStateDefinitions,
+                                                              Set<String> userRoles,
+                                                              Predicate<AccessControlList> access) {
+        return super.filterCaseStatesByAccess(caseStateDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public List<CaseEventDefinition> filterCaseEventsByAccess(List<CaseEventDefinition> caseEventDefinitions,
+                                                              Set<String> userRoles,
+                                                              Predicate<AccessControlList> access) {
+        return super.filterCaseEventsByAccess(caseEventDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public CaseViewActionableEvent[] filterCaseViewTriggersByCreateAccess(
+        CaseViewActionableEvent[] caseViewTriggers,
+        List<CaseEventDefinition> caseEventDefinitions,
+        Set<String> userRoles) {
+        return super.filterCaseViewTriggersByCreateAccess(caseViewTriggers,
+            caseEventDefinitions,
+            userRoles);
+    }
```

Comment:
We're not doing anything different here so no need to override the methods?
```suggestion
```

Reply:
removed